src/main/java/io/kestra/plugin/git/PushExecutionFiles.java

```diff
@@ -0,0 +1,284 @@
+    private Property<String> gitDirectory = Property.ofValue("_outputs");
+
+    @Schema(
+        title = "Glob pattern(s) to select execution output files from the working directory",
+        description = "If provided, will match files relative to the execution working directory."
+    )
+    private Object files;
+
+    @Schema(
+        title = "A map of key-value pairs where the key is the filename and the value is the URI of the file to upload.",
+        description = "This should be a map of URI, with the key being the filename that will be upload and the value is the URI." +
+            "This property is intended to be used with the output files of other tasks. Many Kestra tasks, incl. all Downloads tasks, " +
+            "output a map of files so that you can directly pass the output property to this task e.g. " +
+            "[outputFiles in the S3 Downloads task](https://kestra.io/plugins/plugin-aws/tasks/s3/io.kestra.plugin.aws.s3.downloads#outputfiles) " +
+            "or the [files in the Archive Decompress task](https://kestra.io/plugins/plugin-compress/tasks/io.kestra.plugin.compress.archivedecompress#files).",
+        anyOf = {Map.class, String.class}
+    )
+    private Object filesMap;
+
+    @Builder.Default
+    private Property<Boolean> errorOnMissing = Property.ofValue(false);
+
+    @Schema(
+        title = "Git commit message"
+    )
+    @Override
+    public Property<String> getCommitMessage() {
+        return Optional.ofNullable(this.commitMessage).orElse(new Property<>("Add files from execution {{ execution.id }}"));
+    }
+
+    @Override
+    public Object globs() {
+        return this.files;
+    }
+
+    @Override
+    public Property<String> fetchedNamespace() {
+        return new Property<>("{{ flow.namespace }}");
+    }
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        List<String> renderedGlobs = switch (this.globs()) {
+            case List<?> globList -> ((List<String>) globList).stream().map(throwFunction(runContext::render)).toList();
+            case String globString -> List.of(runContext.render(globString));
+            case null, default -> List.of();
+        };
+
+        Map<Path, Supplier<InputStream>> contentByPath = this.instanceResourcesContentByPath(
+            runContext,
+            runContext.workingDir().path(),
+            renderedGlobs
+        );
+
+
+        if (contentByPath.isEmpty()) {
+            Boolean failIfMissing = runContext.render(this.errorOnMissing).as(Boolean.class).orElse(false);
+            if (failIfMissing) {
+                throw new IllegalArgumentException("No files matched the provided patterns: " + this.files);
+            } else {
+                runContext.logger().info("No files to push, skipping Git operations.");
+                return Output.builder()
+                    .commitId(null)
+                    .commitURL(null)
+                    .files(null)
+                    .build();
```

Comment:
Not sure it's very verbose to just send this will all those null values in case no files was pushed.

+ reformat the block + remove empty line L167